src/main/java/com/minecolonies/coremod/colony/buildings/BuildingBuilder.java

```diff
@@ -165,6 +173,106 @@ public void writeToNBT(@NotNull final NBTTagCompound compound)
     }
 
     /**
+     * Information a bout a resource
+     * - How many are needed to finish the build
+     * - How many are available to the builder
+     * - How many are in the player's inventory (client side only)
+     */
+    public static class BuildingBuilderResource
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Reduce this class from 74 to the maximum allowed 50 or externalize it in a public class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS2972)